### PR TITLE
[terra-list] Add programmatic support for single/multi-select

### PR DESCRIPTION
**Pull request by artpark**
[UXPLATFORM-7595](https://jira2.cerner.com/browse/UXPLATFORM-7595)
===

### Summary
Adds a new prop that configures the correct accessibility settings for single/multi select lists. The values is an enum of `'none'`, `'single-select'`, `'multi-select'`.

### Deployment Link

https://terra-core-deployed-pr-#.herokuapp.com/

### Testing
Testing done. Tests documented [here](https://cernerprod.sharepoint.com/:x:/s/Terra/EZsW_tccPtxCks1iWVkAVlkB1a8ZZqK_h5yR60sZcxbtmw?e=6zfwpJ)

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by JessieRandle** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please include a CHANGELOG entry for each changed package this PR. Looks like a CHANGELOG is missing for: 

 - terra-toggle-section-header</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ce7a2df14b36396f3832c1feb7aa692c2f7aaf67
</p>

**Comment by JessieRandle** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please include a CHANGELOG entry for each changed package this PR. Looks like a CHANGELOG is missing for: 

 - terra-action-header</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7e684cd9e1559e90fdc68151d6595f5e5b7b5405
</p>

**Comment by JessieRandle** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please include a CHANGELOG entry for each changed package this PR. Looks like a CHANGELOG is missing for: 

 - terra-action-header</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 051210d99a78e32bf6291ada0527218f47f54aec
</p>

File: packages/terra-list/tests/jest/__snapshots__/List.test.jsx.snap [hunk 1]
```diff
@@ -1,212 +1,43 @@
...
-    role="none"
-  >
-    <ul
-      className="list orion-fusion-theme"
-    />
-  </List>
-</ThemeContextProvider>
+<ContextConsumer>
+  <Component />
+</ContextConsumer>
```
Comment (BenBoersma) [inline packages/terra-list/tests/jest/__snapshots__/List.test.jsx.snap:6]: we sure this replacement on the jest test is right?

Comment (artpark) [reply packages/terra-list/tests/jest/__snapshots__/List.test.jsx.snap]: Oops, this was updated accidentally, fixed [9e651a0](https://github.com/cerner/terra-core/pull/3703/commits/9e651a0db2c56fd4025d8c79baf20d07dfc91e24)

**Review by BenBoersma — APPROVED** [review]

**Review by sycombs — APPROVED** [review]

**Comment by scottwilmarth** [comment]
+1 from me :)

**Pull request merged**